pkg/component/etcd/etcd/crd.go

```diff
@@ -9,93 +9,106 @@ import (
 	_ "embed"
 	"fmt"
 
-	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/Masterminds/semver/v3"
+	druidcorev1alpha1 "github.com/gardener/etcd-druid/api/core/v1alpha1"
+	druidcorecrds "github.com/gardener/etcd-druid/api/core/v1alpha1/crds"
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
 
 	"github.com/gardener/gardener/pkg/client/kubernetes"
 	"github.com/gardener/gardener/pkg/component"
 	"github.com/gardener/gardener/pkg/utils/flow"
 	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
 )
 
-var (
-	//go:embed crds/templates/crd-druid.gardener.cloud_etcds.yaml
-	// CRD holds the etcd custom resource definition template
-	CRD string
-	//go:embed crds/templates/crd-druid.gardener.cloud_etcdcopybackupstasks.yaml
-	etcdCopyBackupsTaskCRD string
+type crd struct {
+	client       client.Client
+	applier      kubernetes.Applier
+	crdResources map[string]string
+}
 
-	etcdCRDName                = "etcds.druid.gardener.cloud"
-	etcdCopyBackupsTaskCRDName = "etcdcopybackupstasks.druid.gardener.cloud"
-	crdResources               []string
-)
+// CRDAccess provides methods to manage and get CRDs defined in etcd-druid.
+type CRDAccess interface {
+	component.Deployer
+	CRDGetter
+}
 
-type crd struct {
-	client  client.Client
-	applier kubernetes.Applier
+// CRDGetter provides methods to get CRDs defined in etcd-druid.
+type CRDGetter interface {
+	// GetCRDYaml returns the YAML of the CRD with the given name.
+	// An error is returned if no CRD is found with the given name.
+	GetCRDYaml(name string) (string, error)
+	// GetCRD returns the CRD with the given name.
+	// An error is returned if no CRD is found with the given name.
+	GetCRD(name string) (*apiextensionsv1.CustomResourceDefinition, error)
 }
 
-func init() {
-	crdResources = append(crdResources, CRD, etcdCopyBackupsTaskCRD)
+// NewCRDGetter creates a new CRDGetter.
+func NewCRDGetter(k8sVersion *semver.Version) (CRDGetter, error) {
+	return NewCRD(nil, nil, k8sVersion)
 }
 
-// NewCRD can be used to deploy the CRD definitions for Etcd and EtcdCopyBackupsTask.
-func NewCRD(c client.Client, applier kubernetes.Applier) component.Deployer {
-	return &crd{
-		client:  c,
-		applier: applier,
+// NewCRD can be used to deploy and/or retrieve the CRD definitions for all CRDs defined by etcd-druid.
+func NewCRD(c client.Client, applier kubernetes.Applier, k8sVersion *semver.Version) (CRDAccess, error) {
+	crdResources, err := getEtcdCRDS(k8sVersion)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get etcd-druid CRDs for Kubernetes version: %s:%w", k8sVersion, err)
```

Comment:
Cosmetic nit (suggesting the same as my previous suggestion in https://github.com/gardener/gardener/pull/11545#discussion_r1998618000):

```suggestion
		return nil, fmt.Errorf("failed to get etcd-druid CRDs for Kubernetes version %s: %w", k8sVersion, err)
```